### PR TITLE
[transaction stream] add the external grpc to fullnode.

### PR DESCRIPTION
**Pull request by larry-aptos**
### Description
* Add external grpc to fullndoe, which supports `transactions_count`. 

### Test Plan

Tested locally, looks good. Also the disconnection is handled gracefully. 

with `dev_mode` off:
  
image

with `dev_mode` on:

image

### CODE REVIEW

File: ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs [hunk 1]
```diff
@@ -38,6 +42,14 @@ pub struct FullnodeDataService {
     pub output_batch_size: u16,
 }
 
+// TODO: move it to data service.
```
Comment (bowenyang007) [inline ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs:45]: why?

File: ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs [hunk 2]
```diff
@@ -63,6 +75,12 @@ pub fn bootstrap(
     runtime.spawn(async move {
         let context = Arc::new(Context::new(chain_id, db, mp_sender, node_config));
         let server = FullnodeDataService {
+            context: context.clone(),
+            processor_task_count,
+            processor_batch_size,
+            output_batch_size,
+        };
```
Comment (bowenyang007) [inline ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs:82]: can you call this test and switch based on whether the dev flag is turned on?

File: ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs [hunk 3]
```diff
@@ -38,6 +42,14 @@ pub struct FullnodeDataService {
     pub output_batch_size: u16,
 }
 
+// TODO: move it to data service.
+pub struct RawDataService {
```
Comment (bowenyang007) [inline ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs:46]: rename this to DataServiceForTest

File: ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs [hunk 4]
```diff
@@ -205,3 +224,93 @@ pub fn get_status(
         chain_id: ledger_chain_id as u32,
     }
 }
+
+/// External service on the fullnode is for testing/local development only.
+/// Performance is not optimized, e.g., single-threaded.
+#[tonic::async_trait]
+impl RawData for RawDataService {
```
Comment (bowenyang007) [inline ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs:231]: let's extract this out to a different function?

File: ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs [hunk 5]
```diff
@@ -38,6 +42,14 @@ pub struct FullnodeDataService {
     pub output_batch_size: u16,
 }
 
+// TODO: move it to data service.
+pub struct RawDataService {
+    pub context: Arc<Context>,
+    pub processor_task_count: u16,
+    pub processor_batch_size: u16,
+    pub output_batch_size: u16,
```
Comment (banool) [inline ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs:50]: These are identical to the fields in FullnodeDataService, can you either just use the same struct or extract the common stuff into another struct?

File: ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs (hunk 2 above)
Comment (banool) [reply ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs:82]: What do you mean by dev flag?

File: ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs (hunk 1 above)
Comment (larry-aptos) [reply ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs]: removed.

File: ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs (hunk 2 above)
Comment (larry-aptos) [reply ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs]: added `dev_mode` flag.

File: ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs (hunk 3 above)
Comment (larry-aptos) [reply ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs]: done.

File: ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs (hunk 4 above)
Comment (larry-aptos) [reply ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs]: done.

File: ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs (hunk 5 above)
Comment (larry-aptos) [reply ecosystem/indexer-grpc/indexer-grpc-fullnode/src/runtime.rs]: done.

**Review by banool — APPROVED** [review]

**Review by bowenyang007 — APPROVED** [review]

**Pull request merged**